src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -9,91 +10,330 @@ import {
+  } else if (sku === CosmosD16s) {
+    return "CosmosD16Details";
+  } else if (sku === CosmosD32s) {
+    return "CosmosD32Details";
+  }
+  return "Not Supported Yet";
+};
+
+const onSKUChange = (newValue: InputType, currentValues: Map<string, SmartUiInput>): Map<string, SmartUiInput> => {
+  currentValues.set("sku", { value: newValue });
+  currentValues.set("skuDetails", {
+    value: { textTKey: getSKUDetails(`${newValue.toString()}`), type: DescriptionType.Text } as Description,
+  });
+  currentValues.set("costPerHour", { value: costPerHourValue });
+  return currentValues;
+};
+
+const onNumberOfInstancesChange = (
+  newValue: InputType,
+  currentValues: Map<string, SmartUiInput>
+): Map<string, SmartUiInput> => {
+  currentValues.set("instances", { value: newValue });
+  currentValues.set("warningBanner", {
+    value: { textTKey: "WarningBannerOnUpdate" } as Description,
+    hidden: false,
+  });
+
+  return currentValues;
+};
 
 const onEnableDedicatedGatewayChange = (
   newValue: InputType,
-  currentState: Map<string, SmartUiInput>
+  currentValues: Map<string, SmartUiInput>,
+  baselineValues: ReadonlyMap<string, SmartUiInput>
 ): Map<string, SmartUiInput> => {
-  const sku = currentState.get("sku");
-  const instances = currentState.get("instances");
-  const isSkuHidden = newValue === undefined || !(newValue as boolean);
-  currentState.set("enableDedicatedGateway", { value: newValue });
-  currentState.set("sku", { value: sku.value, hidden: isSkuHidden });
-  currentState.set("instances", { value: instances.value, hidden: isSkuHidden });
-  return currentState;
+  currentValues.set("enableDedicatedGateway", { value: newValue });
+  const dedicatedGatewayOriginallyEnabled = baselineValues.get("enableDedicatedGateway")?.value as boolean;
+  if (dedicatedGatewayOriginallyEnabled === newValue) {
+    currentValues.set("sku", baselineValues.get("sku"));
+    currentValues.set("instances", baselineValues.get("instances"));
+    currentValues.set("skuDetails", baselineValues.get("skuDetails"));
+    currentValues.set("costPerHour", baselineValues.get("costPerHour"));
+    currentValues.set("warningBanner", baselineValues.get("warningBanner"));
+    currentValues.set("connectionString", baselineValues.get("connectionString"));
+    return currentValues;
+  }
+
+  currentValues.set("warningBanner", undefined);
+  if (newValue === true) {
+    currentValues.set("warningBanner", {
+      value: {
+        textTKey: "WarningBannerOnUpdate",
+        link: {
+          href: "https://docs.microsoft.com/en-us/azure/cosmos-db/introduction",
+          textTKey: "DedicatedGatewayPricing",
+        },
+      } as Description,
+      hidden: false,
+    });
+  } else {
+    currentValues.set("warningBanner", {
+      value: {
+        textTKey: "WarningBannerOnDelete",
+        link: {
+          href: "https://docs.microsoft.com/en-us/azure/cosmos-db/introduction",
+          textTKey: "DeprovisioningDetailsText",
+        },
```

Comment:
suggest: for slightly improved readability:


value  {}
if (newValue == true)
{
value = ...
}
else
{
vaue = ..
}
currentValues.set("warningBanner", value, hidden:false)